prov/sockets/include/sock.h

```diff
@@ -869,6 +869,8 @@ struct sock_pe_entry {
 	struct dlist_entry ctx_entry;
 	struct ofi_ringbuf comm_buf;
 	size_t cache_sz;
+
+	ofi_atomic32_t completion_reported;
```

Comment:
I think the sock_pe_entry is in need of a state machine overall.  :(  But to keep this solution simple for now... I don't think this requires an atomic.  The pe_entry must be protected against parallel access, and I think is in all cases.  This can be dropped back to u8 to match the other state variables (is_complete, mr_checked)